File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        for(int i=n;i<=100;i++){
            int mul=1;
            for(int j=i;j>0;j/=10){
                int r=j%10;
                mul*=r;
            }
            if(mul%t==0) return i;
        }
        return 0;
    }
};

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        for i in range(n,101):
            mul=1
            w=i
            while w>0:
                x=w%10
                mul*=x
                w=w//10
            if mul%t==0:
                return i
        return 0